prow/entrypoint/run_test.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package entrypoint
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/pod-utils/wrapper"
+)
+
+func TestOptions_Run(t *testing.T) {
+	var testCases = []struct {
+		name           string
+		args           []string
+		timeout        time.Duration
+		gracePeriod    time.Duration
+		expectedLog    string
+		expectedMarker string
+		expectedErr    bool
+	}{
+		{
+			name:           "successful command",
+			args:           []string{"sh", "-c", "exit 0"},
+			expectedLog:    "",
+			expectedMarker: "0",
+			expectedErr:    false,
+		},
+		{
+			name:           "successful command with output",
+			args:           []string{"echo", "test"},
+			expectedLog:    "test\n",
+			expectedMarker: "0",
+			expectedErr:    false,
+		},
+		{
+			name:           "unsuccessful command",
+			args:           []string{"sh", "-c", "exit 12"},
+			expectedLog:    "",
+			expectedMarker: "12",
+			expectedErr:    true,
+		},
+		{
+			name:           "unsuccessful command with output",
+			args:           []string{"sh", "-c", "echo test && exit 12"},
+			expectedLog:    "test\n",
+			expectedMarker: "12",
+			expectedErr:    true,
+		},
+		{
+			name:           "command times out",
+			args:           []string{"sleep", "10"},
+			timeout:        1 * time.Millisecond,
```

Comment:
Nanosecond? :)

Reply:
I'm weary of that one, only ever seen it cause weird issues in CI with timing